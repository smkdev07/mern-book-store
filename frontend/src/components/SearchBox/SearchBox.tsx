import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';

import { Form, Button } from 'react-bootstrap';

const SearchBox: React.FC = () => {
  const [searchTerm, setSearchTerm] = useState('');
  const history = useHistory();

  const submitHandler = (event: React.FormEvent<HTMLElement>) => {
    event.preventDefault();
    if (searchTerm.trim()) {
      history.push(`/search/${searchTerm}`);
    } else {
      history.push('/');
    }
  };

  return (
    <Form onSubmit={submitHandler} inline>
      <Form.Control
        type="text"
        name="searchTerm"
        placeholder="Search Books..."
        onChange={(event) => setSearchTerm(event.target.value)}
        className="mr-sm-2 ml-sm-5"
      />
      <Button type="submit" variant="outline-success" className="p-2">
        Search
      </Button>
    </Form>
  );
};

export default SearchBox;
