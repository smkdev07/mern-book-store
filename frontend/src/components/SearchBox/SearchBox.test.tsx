import React from 'react';
import { shallow } from 'enzyme';

import SearchBox from './SearchBox';

describe('<SearchBox />', () => {
  it('renders the <SearchBox /> component', () => {
    const wrapper = shallow(<SearchBox />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the <SearchBox /> component with <Form> component', () => {
    const wrapper = shallow(<SearchBox />);
    const form = wrapper.find('Form');
    expect(form).toHaveLength(1);
  });

  it('renders the <SearchBox /> component with <FormControl type="text" /> component', () => {
    const wrapper = shallow(<SearchBox />);
    const formControl = wrapper.find('FormControl').first();
    const formControlType = formControl.prop('type');
    expect(formControlType).toBe('text');
  });

  it('renders the <SearchBox /> component with <Button> component with Label as Search', () => {
    const wrapper = shallow(<SearchBox />);
    const button = wrapper.find('Button').first().text();
    expect(button).toContain('Search');
  });
});
