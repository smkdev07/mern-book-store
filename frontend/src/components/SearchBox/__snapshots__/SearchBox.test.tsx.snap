// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> renders the <SearchBox /> component 1`] = `ShallowWrapper {}`;
